internal/graph/check.go

```diff
@@ -450,6 +450,19 @@ func (c *LocalChecker) ResolveCheck(
 	object := req.GetTupleKey().GetObject()
 	relation := req.GetTupleKey().GetRelation()
 
+	userObject, userRelation := tuple.SplitObjectRelation(req.GetTupleKey().GetUser())
+	if userRelation != "" {
+		if object == userObject && relation == userRelation {
+			return &ResolveCheckResponse{
+				Allowed: true,
+				ResolutionMetadata: &ResolutionMetadata{
+					Depth:               defaultResolveNodeLimit,
```

Comment:
```suggestion
					Depth:               req.GetResolutionMetadata().Depth,
```

since you can't be sure that you encountered the tupleKey at the first call